elastic/kibana
# Resolves UI Breaks from Malformed Roles

Resolves #124808 where malformed Elasticsearch roles cause Kibana users and roles UIs to not display correctly.

### Overview

This PR adds a catch for exceptions thrown in deserialization of Elasticsearch roles during transformation into Kibana format. This provides an additional case to set the transform success flag to false and return from the operation gracefully when a malformed role is encountered, keeping the Kibana Roles and Users UI's intact.

This PR also adds an additional logger parameter to the transform functions, allowing the details of a thrown exception to be logged where previously only a generic message would appear to the Kibana user.

Add "feature_securitySolutionCases.a;;" to a user role
Apply the user role via API
Try to go to Management > Security > Roles or to Management > Security > Users
An error will be illustrated bottom right of the screen

### Testing

1. Use the Kibana Dev Console to add a 'test' role with a malformed privilege.
>PUT /_security/role/test
{
&emsp;"cluster": [],
&emsp;"indices": [],
&emsp;"applications": [
&emsp;{
&emsp;&emsp;"application": "kibana-.kibana",
&emsp;&emsp;"privileges": [
&emsp;&emsp;"feature_securitySolutionCases.a;;"
&emsp;&emsp;],
&emsp;&emsp;"resources": [
&emsp;&emsp;"*"
&emsp;&emsp;]
&emsp;}
&emsp;],
&emsp;"run_as": [],
&emsp;"metadata": {},
&emsp;"transient_metadata": {
&emsp;"enabled": true
&emsp;}
}
2. Verify the Stack Management > Security > Users screen renders and operates without issue. 
3. Verify the Stack Management > Security > Roles screen renders and operates without issue. 
4. Verify the new 'test' role is displayed and can be selected.
5. Select the 'test' role. Verify that in the Kibana section of the Edit Role UI a message is displayed informing the use that the role is malformed.
<img width="1428" alt="Screen Shot 2022-05-11 at 8 33 04 AM" src="https://user-images.githubusercontent.com/103939324/167783886-22675011-cc56-45a3-860d-6c0307e6345c.png">
6. Delete the 'test' role.
